src/Xrefcheck/Config/Default.hs

```diff
@@ -56,6 +56,10 @@ networking:
   # a "429 Too Many Requests" response.
   maxRetries: 3
 
+  # How many attempts to retry an external link after disconnect
+  # due to the timeout.
+  maxTimeoutRetries: 1
```

Comment:
I would elaborate in description of this more, currently it sounds like something related solely to timeouts and not to 429s.

Reply:
And I would mention the motivation for it, which exactly case this serves for

Reply:
Good, but this still mentions only what we do and not why, and the user may miss what's the point behind such complexity.

Instead of the last 3 lines I would just add, in order not to repeat mostly the same information:

```
  # Sometimes, after we exceeded rate limits, the site may stop
  # responding to our requests, and this option helps to workaround
  # that.
```

Reply:
I tried to rewrite two config fields' descriptions from scratch, unfortunately it's best that I'm able to produce. If that version is not acceptable, could you please write these descriptions yourself?

Reply:
> it's best that I'm able to produce. If that version is not acceptable, let's just put your version there

By the way, from the standpoint of an efficient review-edit cycle, very good if you put it clearly; oftentimes I observed how such reviewer's nitpicking turned into infinite dialogs. I saw that you can write good text covering all the necessary points, and here it's rather me who looks at things from a different angle and want accents to be placed differently.

Now coming to your edit itself:

> Total number of allowed 429 errors and timeouts
> when querying an external link.
> On other errors xrefcheck fails immediately, without retrying.

Since the reason for timeouts handling is not obvious at first, I would leave the first variant + extend it:

```
  # How many attempts to retry an external link after getting
  # a "429 Too Many Requests" response.
  # Timeouts may also be accounted here, see the description
  # of `maxTimeoutRetries` field.
  
  # If a site once responded with 429 error code, subsequent
  # request timeouts will also be treated as hitting the site's 
  # rate limiter and result in retry attempts, unless the
  # maximum retries number has been reached.
  #
  # On other errors xrefcheck fails immediately, without retrying.
```

> Querying a given domain that ever returned 429 before,
> this defines how many timeouts are allowed.
> For such domains, xrefcheck considers timeouts as possible
> 429 errors, and for other domains timeout immediately
> results in error, assuming that this site is not working at all.

I would rephrase it as follows:

```
Querying a given domain that ever returned 429 before,
this defines how many timeouts are allowed during retries.

For such domains, timeouts likely mean hitting the rate limiter, 
and so xrefcheck considers timeouts in the same way as 429 errors.

For other domains, a timeout results in a respective error, no retry
attempts will be performed. Use `externalRefCheckTimeout` option
to increase the time after which timeout is declared.

This option is similar to `maxRetries`, the difference is that
this `maxTimeoutRetries` option limits only the number of retries
caused by timeouts, and `maxRetries` limits the number of retries
caused both by 429s and timeouts.
```

Reply:
Thank you very much @Martoon-00

Reply:
Thanks for applying the update :+1:

Reply:
@aeqz If you have thoughts how to improve this further, please comment here too.

Reply:
I think that it is clear now 👍